tmt/steps/report/junit.py

```diff
@@ -58,40 +42,170 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
             f"Malformed duration '{duration}' ({error}).")
 
 
+class ResultsContext:
+    """
+    A class which keeps the results context (especially the result summary) for
+    JUnit template.
+    """
+
+    def __init__(self, results: list[tmt.Result]) -> None:
+        self._results = results
+
+    def __iter__(self) -> Iterator[tmt.Result]:
+        """ Possibility to iterate over results by iterating an instance """
+        return iter(self._results)
+
+    def __len__(self) -> int:
+        """ Returns the number of results """
+        return len(self._results)
+
+    @functools.cached_property
+    def executed(self) -> list[tmt.Result]:
+        """ Returns results of all executed tests """
+        return [r for r in self._results if r.result != ResultOutcome.INFO]
+
+    @functools.cached_property
+    def skipped(self) -> list[tmt.Result]:
+        """ Returns results of skipped tests """
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self) -> list[tmt.Result]:
+        """ Returns results of failed tests """
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self) -> list[tmt.Result]:
+        """ Returns results of tests with error/warn outcome """
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self) -> int:
+        """ Returns the total duration of all tests in seconds """
+        return sum([duration_to_seconds(r.duration) or 0 for r in self._results])
+
+
 def make_junit_xml(
-        report: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]'
-        ) -> 'junit_xml.TestSuite':
-    """ Create junit xml object """
-    junit_xml = import_junit_xml()
+        report: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]',
+        flavor: str,
```

Comment:
* most likely a nitpicking, but please, consider `phase` instead of `report`. I tried to use it throughout the code to represent plugin instances, which is the case here as well.
* and if `default` would be the default value of `flavor`, which would make sense, right, then the call in `polarion` plugin could be even simpler :)

On the other hand, given the note in `polarion`, it makes sense to keep the `flavor='default'` there even if `flavor` would have a default value, because this argument will change when `polarion` flavor arrives. But, I'd still argue that `flavor: str = DEFAULT_FLAVOR` would be very much due.

Reply:
Changed the `report -> phase` and I added the default value to `flavor` argument. Thanks.